proxy/proxy.go

```diff
@@ -1484,6 +1484,11 @@ func (p *Proxy) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 			ctx.Logger().Errorf("Failed to set write deadline: %v", e)
 		}
 	}
+	if sbf, ok := ctx.StateBag()[filters.LimitConcurrency]; ok {
+		if f, ok := sbf.(func()); ok {
+			defer f()
+		}
+	}
```

Comment:
Why do we need to defer it here instead of just decrementing on Response?

Reply:
Because if the backend is fully asynchronous we need to decrement after copystream (after the full body was sent to client )